plugins/fetcher.cc

```diff
@@ -124,7 +172,7 @@ struct FetcherFunction: public ValueFunctionT<FetcherArgs, FetcherOutput> {
         return result;
     }
 
-    FetcherFunctionConfig functionConfig;
+    mutable FetcherFunctionConfig functionConfig;
```

Comment:
No, this should remain const.  You can add a mutable CV and mutex to this class.